config/config.go

```diff
@@ -0,0 +1,181 @@
+	config.FilePath = path
+	return &config
+
+}
+
+func validateFilePath(path string) error {
+	_, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return fmt.Errorf("file does not exist: %s", path)
+	} else if err != nil {
+		return fmt.Errorf("error accessing file: %s", err)
+	}
+	return nil
+}
+
+func ParseHour(timeString string) int {
+	hour, _ := strconv.Atoi(timeString[:len(timeString)-2])
+	if strings.HasSuffix(timeString, "PM") && hour != 12 {
+		hour += 12
+	}
+	if strings.HasSuffix(timeString, "AM") && hour == 12 {
+		hour = 0
+	}
+	return hour
+}
+func validateKeywords(keywords string) ([]string, error) {
+	// Split the keywords string into a slice
+	keywordsSlice := strings.Split(keywords, ",")
+
+	// Create a map to check for duplicates
+	keywordsMap := make(map[string]bool)
+
+	// Create a slice to store the validated keywords
+	validatedKeywords := []string{}
+
+	for _, keyword := range keywordsSlice {
+		// Trim spaces and convert to lowercase for case-insensitive comparison
+		trimmedKeyword := strings.TrimSpace(strings.ToLower(keyword))
+
+		if trimmedKeyword == "" {
+			// Skip empty keywords
+			continue
+		}
+
+		// Check for duplicates
+		if _, exists := keywordsMap[trimmedKeyword]; !exists {
+			keywordsMap[trimmedKeyword] = true
+			validatedKeywords = append(validatedKeywords, trimmedKeyword)
+		}
+	}
+
+	if len(validatedKeywords) == 0 {
+		return nil, fmt.Errorf("no valid keywords provided")
+	}
+
+	return validatedKeywords, nil
+}
+
+func validatePostcode(postcode string) error {
+	if len(postcode) != 5 {
+		return fmt.Errorf("postcode must be 5 digits long")
+	}
+
+	for _, char := range postcode {
+		if !unicode.IsDigit(char) {
+			return fmt.Errorf("postcode must only contain digits")
+		}
+	}
+
+	return nil
+}
+
+func validateTimeWindow(timeWindow string) error {
+	// Split the timeWindow string into start and end times
+	times := strings.Split(timeWindow, "-")
+	if len(times) != 2 {
+		return fmt.Errorf("time window must be in format 'start-end'")
+	}
+
+	// Trim spaces
+	startTimeStr := strings.TrimSpace(times[0])
+	endTimeStr := strings.TrimSpace(times[1])
+
+	// Check that both times contain either "AM" or "PM"
+	if !(strings.Contains(startTimeStr, "AM") || strings.Contains(startTimeStr, "PM")) ||
+		!(strings.Contains(endTimeStr, "AM") || strings.Contains(endTimeStr, "PM")) {
+		return errors.New("both start and end times must specify either 'AM' or 'PM'")
+	}
+
+	// Validate start time
+	if _, err := time.Parse("3PM", startTimeStr); err != nil {
+		return fmt.Errorf("invalid start time: %v", err)
+	}
+
+	// Validate end time
+	if _, err := time.Parse("3PM", endTimeStr); err != nil {
+		return fmt.Errorf("invalid end time: %v", err)
+	}
+
+	startTime, err := time.Parse("3PM", startTimeStr)
+	if err != nil {
+		return fmt.Errorf("invalid start time: %v", err)
+	}
+	endTime, err := time.Parse("3PM", endTimeStr)
+	if err != nil {
+		return fmt.Errorf("invalid end time: %v", err)
+	}
+
```

Comment:
Кажется как будто лишняя валидация
```suggestion
	startTime, err := time.Parse("3PM", startTimeStr)
	if err != nil {
		return fmt.Errorf("invalid start time: %w", err)
	}
	endTime, err := time.Parse("3PM", endTimeStr)
	if err != nil {
		return fmt.Errorf("invalid end time: %w", err)
	}
```
Ошибку в этом случае можно завернуть - это позволяет понять как выглядела изначальная ошибка

Reply:
исправил метод